hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculator.java

```diff
@@ -192,6 +202,45 @@ public void updateNodes(final Instant consensusTime) {
                         NO_CUSTOM_FEES, NO_OTHER_SIDE_EFFECTS, END_OF_STAKING_PERIOD_CALCULATIONS_MEMO));
     }
 
+    /**
+     * Scales up the weight of the node to the range [minStake, maxSTakeOfAllNodes] from the consensus weight range [0, 500].
+     * @param weight weight of the node
+     * @param minStake min stake of the node
+     * @param maxStakeOfAllNodes real max stake of all nodes computed by taking max(stakeOfNode1, stakeOfNode2, ...)
+     * @return scaled weight of the node
+     */
+    long scaleUpWeightToStake(
+            final int weight, final long minStake, final long maxStakeOfAllNodes, final long totalStakeOfAllNodes) {
+        final var zeroStake = 0;
+        final var newMin = minStake;
+        final var newMax = maxStakeOfAllNodes;
+        // If zero stake return zero
+        if (weight == zeroStake) {
+            return zeroStake;
+        } else {
+            // This should never happen, but if it does, return zero
+            if (totalStakeOfAllNodes == 0) {
+                return 0;
+            }
+            final var oldMinWeight = 1;
+            // Since we are calculating weights based on the real stake values, we need to consider
+            // the real max Stake and not theoretical max stake of nodes.
+            // Otherwise, on a one node network with a stake < maxStake where we assign a weight of 500,
+            // the scaled stake value will be greater than its real stake.
+            final var oldMaxWeight = BigInteger.valueOf(maxStakeOfAllNodes)
+                    .multiply(BigInteger.valueOf(500))
+                    .divide(BigInteger.valueOf(totalStakeOfAllNodes))
+                    .longValue();
+            // Otherwise compute the interpolation of the weight in the range [minStake, maxStake]
+            return BigInteger.valueOf(newMax)
+                    .subtract(BigInteger.valueOf(newMin))
+                    .multiply(BigInteger.valueOf(weight - oldMinWeight))
+                    .divide(BigInteger.valueOf(oldMaxWeight - oldMinWeight))
+                    .add(BigInteger.valueOf(newMin))
+                    .longValue();
```

Comment:
This doesn't seem like a step function as discussed earlier today. Do these numbers need to be put into 'buckets' of results via some min/max calculation?

Reply:
They are in buckets as in some range of values in tinybars that correspond to certain weight, will be scaled down to some weight. 

For example, in this unit [test](https://github.com/hashgraph/hedera-services/blob/cd3161d1aa268d376758a759620a75856ffdaa2d/hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/ledger/accounts/staking/EndOfStakingPeriodCalculatorTest.java#L130) , stakeInBetween1, stakeInBetween2 will both be scaled down to `119_779_900_327L` because they both get same weight.